src/provider/model/Model.ts

```diff
@@ -175,13 +175,11 @@ export class Model {
         }, []);
     }
 
-    private onWindowSeen(identity: Identity): void {
-        this.getOrCreateExpectedWindow(identity);
-    }
-
-    private async onWindowCreated(identity: Identity, manifestUrl: string): Promise<void> {
+    private async onWindowSeen(identity: Identity): Promise<void> {
         const apps = await this._directory.getAllApps();
-        const appInfoFromDirectory = apps.find(app => app.manifest.startsWith(manifestUrl));
+        const appInfoFromDirectory = apps.find((app) => {
+            return app.appId === identity.uuid || checkCustomConfigField(app, CustomConfigFields.OPENFIN_APP_UUID) === identity.uuid;
```

Comment:
Do we definitely want to check both `app.appId` and the custom fields. My intuition is that if a directory entry has the `appUuid` custom field, then we should ignore the `app.appId`. As it is, there's potentially two different apps that would get matched to this directory entry - the one that matches `app.appId`, and the one that matches the custom field.

Could do as something like 

    return checkCustomConfigField(app, CustomConfigFields.OPENFIN_APP_UUID, app.appId) === identity.uuid;`

where `checkCustomConfigField` (would need renaming I think) uses the custom field if available, or just returns the fallback third param we pass in.

Reply:
makes sense